Source/RapyutaSimulationPlugins/Private/Core/RRCrowdROSController.cpp

```diff
@@ -15,17 +15,19 @@ void ARRCrowdROSController::OnPossess(APawn* InPawn)
     // instantiated yet
     // + InPawn's child class' ros2-related accessories (ROS2 node, sensors, publishers/subscribers)
     //  may have not been fully accessible until now
-    auto* robot = CastChecked<ARRBaseRobot>(InPawn);
-    verify(IsValid(robot->ROS2Interface));
-    ROS2Interface = robot->ROS2Interface;
-    ROS2Interface->Initialize(robot);
+    auto* robotVehicle = GetPawn<ARRBaseRobot>();
+    if(robotVehicle)
+    {
+        robotVehicle->InitROS2Interface();
+    }
 }
 
 void ARRCrowdROSController::OnUnPossess()
 {
-    if (ROS2Interface)
+    auto* robotVehicle = GetPawn<ARRBaseRobot>();
+    if (robotVehicle)
     {
-        ROS2Interface->StopPublishers();
+        robotVehicle->StopROS2Interface();
```

Comment:
Yeah I see, generally an exception (user-oriented) should not be mistaken as a 
design error (dev-oriented).
* We should definitely use `Cast` only for exception but `CastChecked` for the latter like an assert to help detect early error in the design.
* In this specific case, it must **never** be a nullptr essentially since this class (as having its ROS in its name) is meant to work with out ROS2Interface, which requires itself to inherit from `ARRBaseRobot`

Reply:
Controller pauses/unpauses can be happen from blueprint/c++ dynamically as well. Since restarting edtior would slow down develpment, I prefer not to exit editor.

Reply:
Ah I see, so in that case InPawn is null by itself

Reply:
Also `robotVehicle` -> `robot`?

Reply:
Ah I see, so in those cases, InPawn is null by itself, so a log is good.
Besides robotVehicle -> robot?